### PR TITLE
new: matrix(::NCRing, ::MatSpaceElem)

### PR DESCRIPTION
**Pull request by YueRen**
Not sure whether the implementation is okay, but it would be very useful if `matrix(_, _)` would accept a `MatSpaceElem` as second argument:

```
using AbstractAlgebra
R,_ = polynomial_ring(QQ,2)
M = [one(R) one(R); one(R) one(R)]
M = matrix(R,M)
M = matrix(R,M) # this currently does not work
```

I generally convert all matrices into `MatSpaceElem`s to avoid problems such as  https://github.com/Nemocas/AbstractAlgebra.jl/issues/1730,  and `matrix(_, _)` is my go-to method for doing so.

### CODE REVIEW

**Comment by YueRen** [comment]
I'm not sure where to add tests for the function.  I've been looking where tests for `matrix(::NCRing, _)` are, and just added a couple of lines to it.

File: src/Matrix.jl [hunk 1]
```diff
@@ -6443,6 +6443,10 @@ function matrix(R::NCRing, arr::AbstractMatrix{T}) where {T}
    end
 end
 
+function matrix(R::NCRing, arr::MatElem)
+    return map_entries(R,arr)
```
Comment (fingolfin) [inline src/Matrix.jl:6447]:
```suggestion
    return map_entries(R, arr)
```

File: test/generic/Matrix-test.jl [hunk 2]
```diff
@@ -325,7 +325,9 @@ end
 
    M = matrix(ZZ, M1)
    N = matrix(ZZ, N1)
-   K = matrix(ZZ, K1)
+
+   @test matrix(ZZ,M1) == M1
```
Comment (fingolfin) [inline test/generic/Matrix-test.jl:329]:
```suggestion
   @test matrix(ZZ, M1) == M1
```

File: test/generic/Matrix-test.jl [hunk 3]
```diff
@@ -325,7 +325,9 @@ end
...
+   @test matrix(ZZ,N1) == N1
```
Comment (fingolfin) [inline test/generic/Matrix-test.jl:330]:
```suggestion
   @test matrix(ZZ, N1) == N1
```

File: test/generic/Matrix-test.jl [hunk 4]
```diff
@@ -325,7 +325,9 @@ end
 
    M = matrix(ZZ, M1)
    N = matrix(ZZ, N1)
```
Comment (fingolfin) [inline test/generic/Matrix-test.jl:327]:
```suggestion
   N = matrix(ZZ, N1)
   K = matrix(ZZ, K1)
```

**Review by fingolfin — APPROVED** [review]

**Comment by fingolfin** [comment]
Fine by me. @fieker also had no complaints during triage today.

**Pull request merged**